components/channel_layout/channel_controller.test.tsx

```diff
@@ -2,99 +2,54 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {shallow} from 'enzyme';
-
-import ChannelController from './channel_controller';
+import {render} from '@testing-library/react';
+
+import ChannelController, {getClassnamesForBody} from './channel_controller';
+
+jest.mock('components/announcement_bar', () => () => <div/>);
```

Comment:
Why do we need to mock all of these now?

Reply:
because rtl doesnot support shallow mocking. it will render all components. But the test objectives in this is not to test all those components since they have tests on themselves in their folders.